### PR TITLE
Add clickable popover backdrop

### PR DESCRIPTION
**Pull request by jazeee**
Fix Add clickable popover backdrop #56

Clicking outside of the popover should close it.
Also, add a title, spell name, and close button to popover.

image

https://user-images.githubusercontent.com/5629800/211223423-56f3ae6b-ebef-4b09-a878-117929c43f9d.mp4

Popover backdrops are a bit tricky in Roll20.
My original attempt was to do `<button class="popoverBackdrop">Click me</button>`
However, this fails in Roll20
```
Uncaught TypeError: Cannot read properties of null (reading 'dataset')
    at HTMLButtonElement.<anonymous> (vtt.bundle.js:formatted:556749:45)
    at HTMLBodyElement.dispatch (jquery-1.9.1.js:3074:9)
    at elemData.handle (jquery-1.9.1.js:2750:28)
```
And looking at the source code:
The error occurs here:
`else if ($(this)[0].offsetParent.dataset.reprowid) {`
which fails due to `offsetParent` being undefined.

Buttons in general, appear to need a parent.

Instead, to work around this, I add a parent div, with `position: fixed`, which solves this issue.

### CODE REVIEW

File: ui/components/popover/popover.scss [hunk 1]
```diff
@@ -17,9 +17,36 @@ button.popoverButton {
...
+    cursor: pointer;
+    button {
+      /**
+       * Unfortunately, the Roll20/jquery setup fails if the button is styled as the popoverBackdrop.
+       * This needs to be a button with a parent, otherwise, we see errors like:
+       * ```
+       * vtt.bundle.js:formatted:556749 Uncaught TypeError: Cannot read properties of null (reading 'dataset')
+       *   at HTMLButtonElement.<anonymous> (vtt.bundle.js:formatted:556749:45)
+       *   at HTMLBodyElement.dispatch (jquery-1.9.1.js:3074:9)
+       *   at elemData.handle (jquery-1.9.1.js:2750:28)
```
Comment (jazeee) [inline ui/components/popover/popover.scss:42]: Popover backdrops are a bit tricky in Roll20.
My original attempt was to do `<button class="popoverBackdrop">Click me</button>`
However, this fails in Roll20
```
Uncaught TypeError: Cannot read properties of null (reading 'dataset')
    at HTMLButtonElement.<anonymous> (vtt.bundle.js:formatted:556749:45)
    at HTMLBodyElement.dispatch (jquery-1.9.1.js:3074:9)
    at elemData.handle (jquery-1.9.1.js:2750:28)
```
And looking at the source code:
The error occurs here:
`else if ($(this)[0].offsetParent.dataset.reprowid) {`
which fails due to `offsetParent` being undefined.

Buttons in general, appear to need a parent.

Instead, to work around this, I add a parent div, with `position: fixed`, which solves this issue.

File: ui/components/popover/popover.scss [hunk 2]
```diff
@@ -17,9 +17,36 @@ button.popoverButton {
...
+       * ```
+       */
+      width: 100%;
+      height: 100%;
+      opacity: 0;
```
Comment (jazeee) [inline ui/components/popover/popover.scss:47]: The button is transparent, but still is clickable.
This hides any oddities or custom button styles.

File: ui/topSection/topSection.html [hunk 3]
```diff
@@ -137,6 +137,9 @@
         </button>
         <input name="attr_chosenPopover" class="sectioncontrol" type="hidden" value="false" />
         <div class="section weightPenalties popoverContainer">
+          <div class="popoverBackdrop">
+            <button name="act_popover_weightPenalties" type="action">Close popover</button>
+          </div>
```
Comment (jazeee) [inline ui/topSection/topSection.html:142]: I'll be following up with the rest of the pug conversions...

**Review by johnlamping — APPROVED** [review]
This is great!

I'll use it tomorrow, as I work on NPCs, and if I don't encounter any critical problems, I think we should offer it to Jay.

**Pull request merged**